apps/lockfile-explorer-web/src/linter.ts

```diff
@@ -0,0 +1,238 @@
+            //   seenSideBySide[dep.name] = dep.version;
+            // }
+          } else {
+            visited.add(dep.resolvedEntry);
+            helper(dep.resolvedEntry);
+          }
+        }
+      }
+      parentStack.pop();
+    }
+
+    helper(projectEntry);
+  }
+
+  const filteredNodes: typeof clusterNodes = {};
+  for (const [key, val] of Object.entries(clusterNodes)) {
+    if (val.duplicateVersions.size === 1) continue;
+    filteredNodes[key] = val;
+  }
+
+  // For each cluster node, go through the allowedConnected list's projects and look for any projects that:
+  // Have the same origin project
+  const filteredNodes2: {
+    [key in string]: {
+      duplicateVersions: Set<string>;
+      allowedConnected: {
+        [key in string]: string[][];
+      };
+      notConnected: {
+        [key in string]: string[][];
+      };
+    };
+  } = {};
+  function getNewFilteredNodes(
+    packageName: string,
+    val: {
+      duplicateVersions: Set<string>;
+      allowedConnected: {
+        [x: string]: string[][];
+      };
+    }
+  ) {
+    // New filtered allowedConnected object
+    const newConnected: typeof val.allowedConnected = {};
+    const notConnected: typeof val.allowedConnected = {};
+    // Look through the current allowedConnected object (which is an object of { [projectName]: parentPath[] }
+    for (const [projectName, parentPaths] of Object.entries(val.allowedConnected)) {
+      let currMax = 0;
+      for (const parentPath of parentPaths) {
+        const rootParent = parentPath[1]; // As parentPath[0] is the current projectName
+        if (!connectedProjectsToClusters[rootParent]) {
+          // package in parent path isn't a project
+          break;
+        } else {
+          // if it is a project, update currMax
+          currMax = Math.max(connectedProjectsToClusters[rootParent][packageName].size, currMax);
+        }
+      }
+      // In this case, currMax can represent two things:
+      // If it is 0, it means there are no other projects further down the chain from this project
+      // Otherwise, if it is > 0, then that number indicates the max number of clusters the child has
+      // If the child of any path has a higher Max than the count for this package, then there is no point in listing this package.
+      if (currMax > connectedProjectsToClusters[projectName][packageName].size || currMax === 0) {
+        newConnected[projectName] = parentPaths;
+      } else {
+        notConnected[projectName] = parentPaths;
+      }
+    }
+    filteredNodes2[packageName] = {
+      ...val,
+      allowedConnected: newConnected,
+      notConnected: notConnected
+    };
+  }
+  for (const [packageName, val] of Object.entries(filteredNodes)) {
+    getNewFilteredNodes(packageName, val);
+  }
+  // console.log('eslint: ', filteredNodes['eslint']);
+
+  // getNewFilteredNodes('eslint', filteredNodes['eslint'])
+
+  console.log('DEBUG MODE');
```

Comment:
Can you use a `Terminal` (from `node-core-library`) instead?

Reply:
Does the `node-core-library` work in the browser? Currently the output is being displayed in the browser console

Reply:
Oh, I didn't realize this was browser code. Disregard.